app/src/test/java/ch/hikemate/app/model/facilities/FacilitiesRepositoryTest.kt

```diff
@@ -0,0 +1,334 @@
+          "amenity": "toilets"
+        }
+      }"""
+  }
+
+  private lateinit var client: OkHttpClient
+  private lateinit var mockCall: Call
+  private lateinit var facilitiesRepository: FacilitiesRepository
+
+  private val callbackCaptor = argumentCaptor<okhttp3.Callback>()
+
+  private val testBoundsNormal = Bounds(46.5, 6.6, 46.51, 6.62)
+
+  private fun buildResponse(request: String): Response {
+    return Response.Builder()
+        .code(200)
+        .message("OK")
+        .body(request.toResponseBody())
+        .protocol(Protocol.HTTP_1_1)
+        .header("Content-Type", "application/json")
+        .request(mock())
+        .build()
+  }
+
+  private fun setupMockResponse(response: Response) {
+    `when`(mockCall.enqueue(callbackCaptor.capture())).then {
+      callbackCaptor.firstValue.onResponse(mockCall, response)
+    }
+  }
+
+  private fun setupNetworkFailure() {
+    `when`(mockCall.enqueue(callbackCaptor.capture())).then {
+      callbackCaptor.firstValue.onFailure(mockCall, IOException("Network failure"))
+    }
+  }
+
+  @Before
+  fun setUp() {
+    client = mock()
+    mockCall = mock()
+
+    facilitiesRepository = FacilitiesRepository(client)
+
+    `when`(client.newCall(any())).thenReturn(mockCall)
+  }
+
+  @Test
+  fun testGetFacilities_integrationTest() =
+      runTest(timeout = 5.seconds) {
+
+        // Calls the real API, not a mocked one
+        val realFacilitiesRepository = FacilitiesRepository(OkHttpClient())
+        realFacilitiesRepository.getFacilities(
+            testBoundsNormal,
+            { routes ->
+              assertNotEquals(routes.size, 0)
+              print(routes)
+            }) {
+              fail("Should not fail")
+            }
+      }
+
+  @Test
+  fun testGetFacilities_callsClient() {
+    `when`(client.newCall(any())).thenReturn(mock())
+    facilitiesRepository.getFacilities(
+        testBoundsNormal,
+        { routes ->
+          assertNotEquals(routes.size, 0)
+          print(routes)
+        }) {
+          fail("Failed to fetch from Overpass API")
+        }
+    verify(client).newCall(any())
+  }
+
+  @Test
+  fun testGetFacilities_andFilterAmenitiesWithNormalInput() {
+
+    val response =
+        buildResponse(VALID_RESPONSE_HEADER + VALID_RESPONSE_ELEMENT + VALID_RESPONSE_FOOTER)
+
+    val callbackCaptor = argumentCaptor<okhttp3.Callback>()
+    val latch = CountDownLatch(1)
+
+    setupMockResponse(response)
+
+    facilitiesRepository.getFacilities(
+        bounds = testBoundsNormal,
+        onSuccess = { facilities ->
+          assertEquals(1, facilities.size)
+          assertEquals(FacilityType.TOILETS, facilities[0].type)
+          latch.countDown()
+          Log.d(
+              "FacilitiesRepositoryTest",
+              "getFacilities with normal parameters succeeded. Got: ${facilities.size} facilities")
+        },
+        onFailure = { fail("getFacilities call failed") })
+
+    verify(mockCall).enqueue(callbackCaptor.capture())
```

Comment:
Warning: same remark as above. Please check the the callback is called with a boolean

Reply:
done